otx/api/usecases/exportable_code/prediction_to_annotation_converter.py

```diff
@@ -419,31 +433,36 @@ def convert_to_annotation(self, predictions: tuple, metadata: Dict[str, Any]) ->
             AnnotationSceneEntity: OTX annotation scene entity object.
         """
         annotations = []
-        for score, class_idx, _, mask in zip(*predictions):
-            mask = mask.astype(np.uint8)
-            contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
-            if hierarchies is None:
-                continue
-            for contour, hierarchy in zip(contours, hierarchies[0]):
-                if hierarchy[3] != -1:
+        height, width, _ = metadata["original_shape"]
+        shape: Union[Polygon, Ellipse]
+        for score, class_idx, box, mask in zip(*predictions):
+            if self.use_ellipse_shapes:
+                shape = convert_bbox_to_ellipse(box[0] / width, box[1] / height, box[2] / width, box[3] / height)
+            else:
+                mask = mask.astype(np.uint8)
+                contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
+                if hierarchies is None:
                     continue
-                if len(contour) <= 2 or cv2.contourArea(contour) < 1.0:
-                    continue
-                contour = list(contour)
-                points = [
-                    Point(
-                        x=point[0][0] / metadata["original_shape"][1],
-                        y=point[0][1] / metadata["original_shape"][0],
-                    )
-                    for point in contour
-                ]
-                polygon = Polygon(points=points)
-                annotations.append(
-                    Annotation(
-                        polygon,
-                        labels=[ScoredLabel(self.labels[int(class_idx) - 1], float(score))],
-                    )
+                for contour, hierarchy in zip(contours, hierarchies[0]):
+                    if hierarchy[3] != -1:
+                        continue
+                    if len(contour) <= 2 or cv2.contourArea(contour) < 1.0:
+                        continue
+                    contour = list(contour)
+                    points = [
+                        Point(
+                            x=point[0][0] / width,
+                            y=point[0][1] / height,
+                        )
+                        for point in contour
+                    ]
+                    shape = Polygon(points=points)
+            annotations.append(
+                Annotation(
+                    shape,
+                    labels=[ScoredLabel(self.labels[int(class_idx) - 1], float(score))],
                 )
+            )
```

Comment:
Multiple shapes should be appended in case of polygons according to original implementation. `shape` in line 459 is being discarded.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/2172/commits/7e79278f46004a54bd5e8595056c96a6bbb7cb25